File: python/3754.concatenate-non-zero-digits-and-multiply-by-sum-i.py
class Solution:
    def sumAndMultiply(self, n: int) -> int:
        nums = []

        while n:
            n, d = divmod(n, 10)

            if d != 0:
                nums.append(d)

        x = 0
        s = 0

        while nums:
            d = nums.pop()
            x *= 10
            x += d
            s += d

        return x * s

File: rust/3754.concatenate-non-zero-digits-and-multiply-by-sum-i.rs
impl Solution {
    pub fn sum_and_multiply(mut n: i32) -> i64 {
        let mut v = Vec::new();

        while n > 0 {
          let d = n % 10;

          if d > 0 {
            v.push(d as i64);
          }

          n /= 10;
        }

        let mut x = 0;
        let mut s = 0;

        while let Some(d) = v.pop() {
          x *= 10;
          x += d;
          s += d;
        }

        x * s
    }
}